main.go

```diff
@@ -128,10 +341,161 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
 		lastFetchedTimestampUintParseBits,
 	)
 	if err != nil {
-		logrus.Errorf("An error occurred parsing last-fetch Unix time string '%v':\n%v", err)
+		return false, stacktrace.Propagate(err, "An error occurred parsing last-fetch Unix time string: '%v'.", lastFetchedUnixTimeStr)
 	}
 	lastFetchedTime := time.Unix(int64(lastFetchedUnixTime), extraNanosecondsToAddToLastFetchedTimestamp)
 	noFetchNeededBefore := lastFetchedTime.Add(fetchGracePeriod)
 
-	return time.Now().After(noFetchNeededBefore)
+	return time.Now().After(noFetchNeededBefore), nil
 }
+
+func getLatestReleaseVersion(repo *git.Repository) (*semver.Version, error) {
+	tagrefs, err := repo.Tags()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while retrieving tags for repository.")
+	}
+
+	// Trim tagrefs and filter for only tags with X.Y.Z version format
+	var allTagSemVers []*semver.Version
+	err = tagrefs.ForEach(func(tagref *plumbing.Reference) error {
+		tagName := tagref.Name().String()
+		tagName = strings.ReplaceAll(tagName, tagsPrefix, "")
+
+		if semverRegex.Match([]byte(tagName)) {
+			tagSemVer, err := semver.StrictNewVersion(tagName)
+			if err != nil {
+				return stacktrace.Propagate(err, "An error occurred while retrieving the following tag: %s.", tagName)
+			}
+			allTagSemVers = append(allTagSemVers, tagSemVer) 
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var latestReleaseTagSemVer *semver.Version
+	if len(allTagSemVers) == 0 {
+		latestReleaseTagSemVer, err = semver.StrictNewVersion(noPreviousVersion)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred creating '%s' semantic version.", noPreviousVersion)
+		}
+	} else {
+		sort.Sort(sort.Reverse(semver.Collection(allTagSemVers)))
+		latestReleaseTagSemVer = allTagSemVers[0]
+	}
+
+	return latestReleaseTagSemVer, nil
+}
+
+func doBreakingChangesExist(changelogFilepath string) (bool, error) {
+	changelogFile, err := os.Open(changelogFilepath)
+	if err != nil {
+		return false, stacktrace.Propagate(err, "An error occurred attempting to open changelog file at provided path. Are you sure '%s' exists?", changelogFilepath)
+	}
+	defer changelogFile.Close()
+
+    scanner := bufio.NewScanner(changelogFile)
+	// Find TBD header
+    for scanner.Scan() {
+		if tbdHeaderRegex.Match(scanner.Bytes()) {
+			break
+		}
+    }
+	if err := scanner.Err(); err != nil {
+        return false, stacktrace.Propagate(err, "An error occurred while scanning for the TBD header in the changelog file at provided path: '%s'.\n", changelogFilepath)
```

Comment:
1) You can replace the literal `TBD` here with the constant from above (first comment), so that if we ever rename this (e.g. `TODO`) then everything will dynamically flow.
2) no need for trailing `\n`
3) same `''`/`:`/`.` thing to resolve